hugegraph-core/src/main/java/com/baidu/hugegraph/backend/cache/RamCache.java

```diff
@@ -50,8 +50,8 @@
     private volatile long expire = 0L;
 
     // NOTE: the count in number of items, not in bytes
-    private final int capacity;
-    private final int halfCapacity;
+    private volatile final int capacity;
+    private volatile final int halfCapacity;
```

Comment:
IMO, since they won't be changed after initialization(with final modifier), so there is no need to set volatile.